tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.py

```diff
@@ -99,47 +132,69 @@ def get_regex_buckets(event):
         raise ExtractFilePathsError(f'KeyError: "{level}[{str(err)}]" is required')
     return regex_buckets
 
-def handler(event, context):            #pylint: disable-msg=unused-argument
+
+def should_exclude_files_type(granule_url: str, exclude_file_types: List[str]) -> bool:
+    """
+    Tests whether or not file is included in {excludeFileTypes}.
+    Args:
+        granule_url: s3 url of granule.
+        exclude_file_types: List of file extensions to exclude from sending to request_files
+    Returns:
+        True if file should be excluded from copy, False otherwise.
+    """
+    for file_type in exclude_file_types:
+        # Returns the first instance in the string that matches .ext or None if no match was found.
+        if re.search(f"^.*{file_type}$", granule_url) is not None:
+            LOGGER.warn(
+                f"The file {granule_url} will not be restored because it matches the excluded file type {file_type}."
+            )
+            return True
+        else:
+            LOGGER.debug(f"File {granule_url} will be restored")
```

Comment:
This current logic causes this message to be logged for every excludeFileType value in the array. You will want to remove the else and move `LOGGER.debug(f"File {granule_url} will be restored")` to the same level as the `return False`. This will allow the message to only occur once.